openhands/server/listen_socket.py

```diff
@@ -92,7 +93,8 @@ async def connect(connection_id: str, environ):
             session_init_args = {**settings.__dict__, **session_init_args}
 
         git_provider_tokens = create_provider_tokens_object(providers_set)
-        if server_config.app_mode != AppMode.SAAS and user_secrets:
+        # Use user_secrets provider tokens if available and not in SAAS mode
+        if server_config.APP_MODE != AppMode.SAAS and user_secrets:
```

Comment:
```suggestion
        if server_config.app_mode != AppMode.SAAS and user_secrets:
```

Pretty sure the class uses lower case